### DISCUSSION TITLE
Project crashes when loading page

### DISCUSSION THREAD

**Discussion by andreasbvillumsen** [pymupdf/PyMuPDF#1867] [Looking for help] [answered]
Hi.

I am experiencing my application crashing on one of our pdf documents, when using FastAPI/Django to upload the pdf via api from PostMan. The pdf document is working correctly, when using hardcoded path to the file.

I am using python version 3.8.9 and PyMuPDF version 1.20.1

**Comment by JorjMcKie** [comment, marked as answer]
I am lacking information about what is really going on technically in your situation. Please be more verbose: what is PostMan, where to are yu uploading, etc.

**Reply by andreasbvillumsen** [reply]
Hi again, sorry about that.

We are making use of FastAPI to create an api endpoint, which is served using uvicorn web server. Postman is an api platform for developers. The purpose of our application is to redact sensitive information, such as emails and phone numbers from any pdf uploaded to it. 

I have been testing our application with a bunch of pdf documents, and it has been working great for most pdf's. We ran into one pdf, which when uploaded would fail on the ninth page. It would not return any error message in the console, hang up the socket in postman, and no longer respond to new requests. 

We have found out that it will halt after running apply_redactions() method, and it will give us "zsh: segmentation fault  python main.py". This is the only error we have seen it give.

When we pass it a local file path to the same pdf document, and don't use the FastAPI python interpreter, the file is successfully redacted.

**Reply by JorjMcKie** [reply]
Thank you for the details. However still hard to get a clue. As it looks now it may be rooted in the special Python you are using.
Any chance to use FastAPI locally to narrow down possible causes?

**Reply by JorjMcKie** [reply]
I read about **_typer_** on their web site - maybe that's a way to enable local usage.

**Reply by andreasbvillumsen** [reply]
We are running the fastapi project locally. From our debugging it appears to be related to images, specifically vector images. It crashes on multiple pages of this document, and it crashes when the apply_redactions method is called.

**Reply by JorjMcKie** [reply]
Try different options to handle images:
I know there is a bug leading to segfaults if images with transparency mask are overlapping redaction rectangles.
Options for image handling (see docu) are: ignore the presence of images, remove any image intersecting a redact, or: build a new, replacing image that is white on overlapping areas. The last one is the default.

So use either `page.apply_redactions(images=option)` where options is either `PDF_REDACT_IMAGES_NONE` or `PDF_REDACT_IMAGES_REMOVE`.

**Reply by JorjMcKie** [reply]
There is issue #1824, which might address also your problem.

**Reply by andreasbvillumsen** [reply]
I have tried using PDF_REDACT_IMAGE_NONE, PDF_REDACT_IMAGE_REMOVE and PDF_REDACT_IMAGE_PIXELS. None of them works.

**Reply by andreasbvillumsen** [reply]
If I remove the apply_redactions method, then it will give me the pdf document with the red annotations, so it must be something happening in the apply_redaction method.

**Reply by JorjMcKie** [reply]
Well then this may be a new issue. If you want, enter a bug report with all accompanying information - i.e. use the template please and provide data (file) that reproduces the problem.

**Reply by andreasbvillumsen** [reply]
Ok I will create an issue. I am unable to send the file in public github, but I have been approved to send it directly to you. Do you have an email I can send the file to?

**Reply by JorjMcKie** [reply]
sure jorj.x.mckie@outlook.de
I might need to consult colleagues of the repo maintainer crew: all of them are employees of the owner company Artifex.
But I would ask for your ok before doing so.

**Reply by JorjMcKie** [reply]
You will also include a minimal reproducer script in the issue description, won't you?

**Reply by andreasbvillumsen** [reply]
Yes that is fine, and you can share the document with your team members. I will create the issue later today or tomorrow, and send pdf to your email as well.
